components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,463 @@
+        }
+    }
+
+    @Test(dataProvider = "dataForDoubleAuthzCodePersisting")
+    public void testGetLatestAuthorizationCodesByTenant(String consumerKey1, String authzCodeID1,
+                                                        String authzCode1,
+                                                        String consumerKey2, String authzCodeID2,
+                                                        String authzCode2) throws Exception {
+
+        AuthzCodeDO authzCodeDO1 = persistAuthorizationCode(consumerKey1, authzCodeID1, authzCode1, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        // Creating an application outside default tenant
+        createApplication(consumerKey2, UUID.randomUUID().toString(), 100);
+        try (Connection connection1 = DAOUtils.getConnection(DB_NAME)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            mockStatic(OAuth2Util.class);
+            mockStatic(IdentityUtil.class);
+            prepareConnection(connection1);
+            when(OAuth2Util.createAuthenticatedUser(anyString(), anyString(), anyString(), anyString())).
+                    thenReturn(authenticatedUser);
+
+            Assert.assertFalse(authorizationCodeDAO.getLatestAuthorizationCodesByTenant(DEFAULT_TENANT_ID).isEmpty());
+            Assert.assertTrue(authorizationCodeDAO.getLatestAuthorizationCodesByTenant(100).isEmpty());
+
+        }
+    }
+
+    protected void createApplication(String consumerKey, String consumerSecret, int tenantId) throws Exception {
+        try (Connection connection = DAOUtils.getConnection(DB_NAME);
+             PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries.ADD_OAUTH_APP)) {
+            prepStmt.setString(1, consumerKey);
+            prepStmt.setString(2, consumerSecret);
+            prepStmt.setString(3, USER_NAME);
+            prepStmt.setInt(4, tenantId);
+            prepStmt.setString(5, UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+            prepStmt.setString(6, APP_NAME);
+            prepStmt.setString(7, "OAuth-2.0");
+            prepStmt.setString(8, CALLBACK);
+            prepStmt.setString(9, "refresh_token urn:ietf:params:oauth:grant-type:saml2-bearer implicit password " +
+                    "client_credentials iwa:ntlm authorization_code urn:ietf:params:oauth:grant-type:jwt-bearer");
+            prepStmt.setLong(10, 3600L);
+            prepStmt.setLong(11, 3600L);
+            prepStmt.setLong(12, 84600L);
+            prepStmt.setLong(13, 3600L);
+            prepStmt.execute();
+            connection.commit();
+        }
+    }
+
+
+    protected void storeIDP() throws Exception {
+        try (Connection connection1 = DAOUtils.getConnection(DB_NAME)) {
+            String sql = "INSERT INTO IDP (TENANT_ID, NAME, UUID) VALUES (1234, 'LOCAL', 5678)";
+            PreparedStatement statement = connection1.prepareStatement(sql);
```

Comment:
need to close this statement (or use try with resource)

Reply:
Addressed.